File: problems/easy/angry_professor/__tests__/test_angry_professor.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "../angry_professor.c"

void test() {
    int k = 3;
    int a[] = {-1, -3, 4, 2};
    int n = 4;

    char* result = angryProfessor(k, a, n);
    if (strcmp(result, "YES") != 0) {
        printf("❌ Test failed. Expected YES, got %s\n", result);
        exit(1);
    }

    printf("✅ C test passed!\n");
}

int main() {
    test();
    return 0;
}

File: problems/easy/angry_professor/__tests__/test_angry_professor.py
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from angry_professor import angryProfessor

def test_angry_professor():
    k = 3
    a = [-1, -3, 4, 2]
    result = angryProfessor(k, a)
    assert result == "YES", f"Expected YES, got {result}"

    print("✅ Python test passed!")

if __name__ == "__main__":
    test_angry_professor()
